src/Features/Core/Portable/SolutionCrawler/IdleProcessor.cs

```diff
@@ -143,9 +144,9 @@ private async Task ProcessAsync()
                         await ExecuteAsync().ConfigureAwait(false);
                     }
                 }
-                catch (OperationCanceledException)
+                catch (Exception e) when (FatalError.ReportAndCatchUnlessCanceled(e))
                 {
-                    // ignore cancellation exception
+                    // In case any error happen during the execution, don't exit the loop and continue to work on the next item.
```

Comment:
this breaks the semantics.  if we get cancellationtoken, this will bubble out and break things.

Reply:
make sure any changes you make here are also made to UnitTestingIdleProcessor.  Thanks!

Reply:
Ah, my bad, will update this